src/Microsoft.AspNetCore.SignalR.Client/ConnectionFactory.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    // REVIEW: Split into Microsoft.AspNetCore.Sockets.Client?
+    public class ConnectionFactory
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly PipelineFactory _pipelineFactory;
+        private readonly ILogger _logger;
+
+        public ConnectionFactory(HttpClient httpClient, PipelineFactory pipelineFactory, ILoggerFactory loggerFactory)
```

Comment:
Hmmmm, it's possible this should be an interface we're passing around. The problem is it gives you back concrete channels.